### PR TITLE
Remove shared web

### PR DESCRIPTION
**Pull request by slorek**
## Description
This removes the `shared-web` Rails engine which was adding friction to the development process. We have already removed it from Submit Cosmetics Product Notifications in https://github.com/UKGovernmentBEIS/beis-opss/pull/1374.

Important: when this is merged, you will no longer need to copy `shared-web` into `vendor` manually. Make sure any changes you're working on separately within `shared-web` are retained when you rebase against this.

The GOV.UK Design System components have been extracted to a new Gem at https://github.com/UKGovernmentBEIS/govuk-design-system-rails.

I also reverted the `sprockets` gem version change which was introduced because there is a breaking change in v4.0.0 which requires a `manifest.js` file which we don't have. This caused failing tests.

## Checklist:

- [x] Automated checks are passing locally.
- [ ] CHANGELOG updated if change is worth telling users about.
### General testing
- [ ] Test without javascript
- [ ] Test on small screen
### Accessibility testing
- [ ] Works keyboard only
- [ ] Tested with one screen reader
- [ ] Zoom page to 400% - content still visible
- [ ] Disable css - does content make sense and appear in a logical order?
- [ ] Passes automated checker (automated in build or manual)

### CODE REVIEW

File: psd-web/Gemfile [hunk 1]
```diff
@@ -38,10 +56,12 @@ group :development, :test do
   gem "rubocop"
   gem "rubocop-performance"
   gem "ruby-debug-ide"
+  gem "sass-rails", "~> 5.0"
   gem "selenium-webdriver"
   gem "simplecov"
   gem "simplecov-console"
   gem "slim_lint"
   gem "solargraph"
+  gem "sprockets", "3.7.2"
```
Comment (StupidCodeFactory) [inline psd-web/Gemfile]: beta 10 works - I know this is related to brakeman complaining about a version under 4.0.0.beta8 that has a vulnerability issue.
But so far the beta10 was working. Has this changed?

File: psd-web/app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
+class SessionsController < ApplicationController
+  skip_before_action :authenticate_user!
+
+  def new
+    redirect_to keycloak_login_url(request.original_fullpath)
+  end
+
+  def signin
+    request_and_store_token(auth_code, params[:request_path])
+    redirect_path = main_app.root_path
```
Comment (StupidCodeFactory) [inline psd-web/app/controllers/sessions_controller.rb]:
```suggestion
```

File: psd-web/app/controllers/sessions_controller.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    redirect_path = params[:request_path] if is_relative(params[:request_path])
```
Comment (StupidCodeFactory) [inline psd-web/app/controllers/sessions_controller.rb]:
```suggestion
    redirect_path = is_relative(params[:request_path]) ? params[:request_path] : root_path
```

File: psd-web/app/controllers/sessions_controller.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    redirect_to redirect_path
+  rescue RestClient::ExceptionWithResponse => e
+    redirect_to keycloak_login_url(params[:request_path]), alert: signin_error_message(e)
+  end
+
+  def logout
+    KeycloakClient.instance.logout(refresh_token)
+    redirect_to main_app.root_path
```
Comment (StupidCodeFactory) [inline psd-web/app/controllers/sessions_controller.rb]:
```suggestion
    redirect_to root_path
```

File: psd-web/app/controllers/sessions_controller.rb [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  def signin_error_message(error)
+    error.is_a?(RestClient::Unauthorized) ? "Invalid email or password." : JSON(error.response)["error_description"]
+  end
+
+  def auth_code
+    params.require(:code)
+  end
+
+  def is_relative(url)
+    url =~ /^\/[^\/\\]/
```
Comment (StupidCodeFactory) [inline psd-web/app/controllers/sessions_controller.rb]: I'd rather have something like 

```ruby 
def allow_redirect?(url)
  ENV['HTTP_HOST'] == URI(url).host
end
```

File: psd-web/app/models/concerns/date_concern.rb [hunk 6]
```diff
@@ -0,0 +1,176 @@
+module DateConcern
```
Comment (StupidCodeFactory) [inline psd-web/app/models/concerns/date_concern.rb]: this entire file seems rather suspicious. I'd love to create a ticket and to examine this and rip it off and re-write it in a legible way.

File: psd-web/app/controllers/sessions_controller.rb (hunk 5 above)
Comment (slorek) [reply psd-web/app/controllers/sessions_controller.rb]: Let's improve it separately, I'm just copying out the files from shared-web as-is in this PR.

File: psd-web/app/controllers/sessions_controller.rb (hunk 4 above)
Comment (slorek) [reply psd-web/app/controllers/sessions_controller.rb]: Good spot thanks.

File: psd-web/Gemfile (hunk 1 above)
Comment (slorek) [reply psd-web/Gemfile]: Here's what happens when I add:

```ruby
gem "sprockets", "= 4.0.0.beta10"
```
to the Gemfile and run tests:

```ruby
Sprockets::Railtie::ManifestNeededError:
  Expected to find a manifest file in `app/assets/config/manifest.js`
  But did not, please create this file and use it to link any assets that need
  to be rendered by your app:

  Example:
    //= link_tree ../images
    //= link_directory ../javascripts .js
    //= link_directory ../stylesheets .css
  and restart your server
```

File: psd-web/app/models/concerns/date_concern.rb (hunk 6 above)
Comment (slorek) [reply psd-web/app/models/concerns/date_concern.rb]: Go ahead! :)

**Comment by slorek** [comment]
Review app URL: https://psd-pr-45-web.london.cloudapps.digital

File: psd-web/Gemfile (hunk 1 above)
Comment (StupidCodeFactory) [reply psd-web/Gemfile]: 🤔 ok don't bother - I have all the assets stuff fixed in my reformat overview page. 
The config wasn't correct in the first place.

**Comment by slorek** [comment]
This was approved by Nasir via Slack.

**Pull request merged**